docstore/drivertest/drivertest.go

```diff
@@ -1163,13 +1177,46 @@ type nativeMinimal struct {
 	B  bool
 	By []byte
 	L  []int
+	A  [2]int
+	A2 [2]int8
+	At ByteArray
 	M  map[string]bool
 	P  *string
 	T  time.Time
 	LF []float64
 	LS []string
 }
 
+// testProto tests encoding/decoding of a document with protocol buffer
+// and pointer-to-protocol-buffer fields.
+func testProto(t *testing.T, _ Harness, coll *ds.Collection) {
```

Comment:
This test would fail before the new `reflect.PtrTo(v.Type()).Implements(protoMessageType)` condition in codec.go's `encode`, because the `Proto` field would not match, and fall through to being encoded as a `map` below. That would fail when decoding since the `map` couldn't be decoded to the `Proto`.

Reply:
I'm fine with this change since it seems you made the user happy, but no one should ever be writing a proto struct directly (as opposed to by pointer).